.eslintrc.cjs

```diff
@@ -0,0 +1,114 @@
+const nodeFiles = [
+  '**/build.dflt.{js,mjs,cjs}',
+  '**/build.{js,mjs,cjs}',
+  '**/config/**',
+  '**/prebuild.{js,mjs,cjs}',
+  '**/prebuild/**',
+  '**/scripts/**',
+  'packages/new-design/lib/**',
+  'sites/backend/**',
+  'sites/*/mdx/**',
+  'sites/*/themes/**',
+]
+const frontendFiles = [
+  '**/components/**',
+  '**/hooks/**',
+  '**/pages/**',
+  '**/page-templates/**',
+  'packages/i18n/**/*.md/*.js',
+]
+const isomorphicFiles = ['packages/core/**', 'plugins/**', 'tests/designs/**', 'tests/plugins/**']
+
+module.exports = {
+  extends: 'eslint:recommended',
+  env: {
+    es2021: true,
+  },
+  // Required when using experimental EcmaScript features
+  parser: '@babel/eslint-parser',
+  parserOptions: {
+    ecmaVersion: 'latest',
+    sourceType: 'module',
+    // Options specific to the Babel parser
+    requireConfigFile: false,
+    babelOptions: {
+      plugins: ['@babel/plugin-syntax-import-assertions'],
+    },
+  },
+  rules: {},
+  overrides: [
+    // Partitioned JavaScript files
+    {
+      files: nodeFiles,
+      env: {
+        node: true,
+      },
+      rules: {
+        'no-console': 'off',
+      },
+    },
+    {
+      files: frontendFiles,
+      excludedFiles: nodeFiles,
+      extends: ['next/core-web-vitals'],
+      env: {
+        // We can be stricter than 'next/core-web-vitals' is
+        node: false,
+      },
+    },
+    {
+      files: isomorphicFiles,
+      excludedFiles: [].concat(nodeFiles, frontendFiles),
+      env: {
+        'shared-node-browser': true,
+      },
+    },
+    // Additional globals for JavaScript files that happen to be CommonJS.
+    // Only allowed in *.cjs files, not *.js files, because we probably want to move towards
+    // a `"type": "module"` future where any CommonJS files would have to have .cjs extensions.
```

Comment:
More background [here](https://nodejs.org/api/packages.html#determining-module-system), though it doesn't directly explain why one might prefer ESM.

Reply:
The bottom line, for developers in this repo: whenever we touch a `*.js`, we'll either need to convert it to ESM in order to lint successfully (and rename it as `*.mjs` if it's in a non-module package, in order to satisfy Node.js), or we'll need to rename it as `*.cjs` (e.g. the `tailwind.config.cjs` for the recalcitrant Tailwind config that @joostdecock mentioned somewhere).